source/common/config/cds_json.cc

```diff
@@ -140,6 +140,9 @@ void CdsJson::translateCluster(const Json::Object& json_cluster,
     cluster.set_type(envoy::api::v2::Cluster::ORIGINAL_DST);
   } else {
     ASSERT(string_type == "sds");
+    if (!eds_config) {
```

Comment:
@mattklein123 , copying from what i mentioned earlier about this change.
"Since cluster_impl_test expects an exception for noSdsConfig test, throwing exception code is added to cds_json.cc." (our previous optional.h throws exception when trying to retrieve value from empty optional.)
Thanks for the review!

Reply:
OK I guess that was a bad error message before. Since you are already in here, can you potentially change the error message to be more useful and change the test to use EXPECT_THROW_WITH_MESSAGE to check it? Thank you!

Reply:
@mattklein123, fixed. please take a look. Thanks!